### PR TITLE
[5.2] useless array access

### PR DESCRIPTION
**Pull request by Denitz**
### Summary of Changes

Useless array access code.

### Testing Instructions

Apply patch

### Actual result BEFORE applying this Pull Request

Bad code.

### Expected result AFTER applying this Pull Request

Good code.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Denitz** [comment]
Please restart drone.

File: libraries/src/Cache/Cache.php [hunk 1]
```diff
@@ -67,8 +67,8 @@ public function __construct($options)
 
         // Overwrite default options with given options
         foreach ($options as $option => $value) {
-            if (isset($options[$option]) && $options[$option] !== '') {
-                $this->_options[$option] = $options[$option];
+            if (isset($value) && $value !== '') {
```
Comment (HLeithner) [inline libraries/src/Cache/Cache.php]:
```suggestion
            if ($value !== null && $value !== '') {
```

File: libraries/src/Cache/CacheController.php [hunk 2]
```diff
@@ -54,8 +54,8 @@ public function __construct($options)
 
         // Overwrite default options with given options
         foreach ($options as $option => $value) {
-            if (isset($options[$option])) {
-                $this->options[$option] = $options[$option];
+            if (isset($value)) {
```
Comment (HLeithner) [inline libraries/src/Cache/CacheController.php]:
```suggestion
            if ($value !== null) {
```

**Review by HLeithner — APPROVED** [review]
A second pair of eyes would be good here

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.2-dev.

**Review by Hackwar — APPROVED** [review]

**Comment by Hackwar** [comment]
Thank you @Denitz for your contribution!

**Pull request merged**